viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+    for i in range(len(user_data["watched"])):
+        avg_rating += user_data["watched"][i]["rating"]
+    if len(user_data["watched"]) == 0:
+        return avg_rating
+    avg_rating /= len(user_data["watched"])
+    return avg_rating
+
+#second swing at get_watched_avg_rating. Which one is better?
+# def get_watched_avg_rating(user_data):
+#     user_watched = user_data["watched"]
+#     avg_rating = 0
+    
+#     for movie in user_watched:
+#         avg_rating += movie["rating"]
+#         print(avg_rating)
+#     if len(user_watched) == 0:
+#         return 0.0
+#     else:
+#         avg_rating /= len(user_watched)
+#         return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    i = 0
+    if len(user_data["watched"]) == 0:
+        return None
+    for key, value in user_data["watched"][i].items():
+        if key == "genre":
+            if value in genre_dict:
+                genre_dict[value] += 1
+            else:
+                genre_dict[value] = 1
+        i += 1

```

Comment:
This loop doesn't iterate over all of the movies in user_data["watched"].  In Python, the for loop statement constructs an iterator from the iterable data that's passed in (in this case `user_data["watched"][i].items()`) before the for loop starts, and then loops over that iterator ([more info if you're interested](https://towardsdatascience.com/python-basics-iteration-and-looping-6ca63b30835c)).  Updating `i` inside the loop here doesn't change what the loop is iterating over.  This for loop will iterate over all the keys of the first dictionary, and add only the genre from that first dictionary to 'genre_dict'.  The code inside here will work if the for loop is rewritten:
```suggestion
    for movie in user_data["watched"]:
        genre = movie["genre"]
        if genre in genre_dict:
            genre_dict[genre] += 1
        else:
            genre_dict[genre] = 1
```

Currently this function only passes the test because the genre of the first movie in the test is the max genre.